### PR TITLE
Make it possible to switch attachment photos in preview section 

### PR DESCRIPTION
**Pull request by anandpeak**

### CODE REVIEW

File: src/modules/common/components/ImageWithPreview.tsx [hunk 1]
```diff
@@ -74,27 +121,122 @@ class ImageWithPreview extends React.Component<Props, State> {
...
+
+            switchedUrl = switchItem(imagesLength - 1);
+          }
+          this.setState({
+            srcUrl: switchedUrl
+          });
+        }
+      }
+    }
+  };
```
Comment (munkhjin0223) [inline src/modules/common/components/ImageWithPreview.tsx]: Unite these functions

File: src/modules/common/components/Attachment.tsx [hunk 2]
```diff
@@ -142,6 +144,9 @@ class Attachment extends React.Component<Props> {
               onLoad={this.onLoadImage}
               alt={attachment.url}
               src={attachment.url}
+              index={this.props.index}
+              switchItem={this.props.switchItem}
+              imagesLength={this.props.imagesLength || 0}
```
Comment (munkhjin0223) [inline src/modules/common/components/Attachment.tsx]: const {  index, switchItem, imagesLength } = this.props;

File: src/modules/common/components/ImageWithPreview.tsx [hunk 3]
```diff
@@ -74,27 +121,122 @@ class ImageWithPreview extends React.Component<Props, State> {
...
+          if (imagesLength > this.state.num + 1) {
+            this.setState({ num: this.state.num + 1 });
+
+            switchedUrl = switchItem(this.state.num) || '';
+          } else {
+            this.setState({ num: 0 });
+
+            switchedUrl = switchItem(0);
+          }
+          this.setState({
```
Comment (munkhjin0223) [inline src/modules/common/components/ImageWithPreview.tsx]: add space

File: src/modules/common/components/ImageWithPreview.tsx [hunk 4]
```diff
@@ -74,27 +121,122 @@ class ImageWithPreview extends React.Component<Props, State> {
...
+    e.stopPropagation();
+
+    const { switchItem, imagesLength } = this.props;
+    const { visible } = this.state;
+
+    let switchedUrl;
+
+    if (visible) {
+      if (switchItem) {
+        if (imagesLength !== undefined) {
```
Comment (munkhjin0223) [inline src/modules/common/components/ImageWithPreview.tsx]: if (visible & ...

File: src/modules/common/components/Attachment.tsx [hunk 5]
```diff
@@ -135,13 +137,17 @@ class Attachment extends React.Component<Props> {
     }
 
     if (attachment.type.startsWith('image')) {
+      const { index, switchItem, imagesLength } = this.props;
       return (
```
Comment (anandpeak) [inline src/modules/common/components/Attachment.tsx]: space

**Pull request closed without merging**